node/spr.go

```diff
@@ -31,27 +33,101 @@ func (d *Pegnetd) GradeS(ctx context.Context, block *factom.EBlock) (graderStake
 	if block.Height >= config.V202EnhanceActivation {
 		ver = 7
 	}
+	if block.Height >= config.PIP18DelegateStakingActivation {
+		ver = 8
+	}
+
+	if ver < 8 {
+		g, err := graderStake.NewGrader(ver, int32(block.Height))
+		if err != nil {
+			return nil, err
+		}
+		for _, entry := range block.Entries {
+			extids := make([][]byte, len(entry.ExtIDs))
+			for i := range entry.ExtIDs {
+				extids[i] = entry.ExtIDs[i]
+			}
+			// allow only top 100 stake holders submit prices
+			stakerRCD := extids[1]
+			if d.Pegnet.IsIncludedTopPEGAddress(stakerRCD) {
+				// ignore bad opr errors
+				err = g.AddSPR(entry.Hash[:], extids, entry.Content)
+				if err != nil {
+					// This is a noisy debug print
+					//logrus.WithError(err).WithFields(logrus.Fields{"hash": entry.Hash.String()}).Debug("failed to add spr")
+				}
+			}
+		}
+		return g.Grade(), nil
+	}
+	return nil, nil
+}
+
+func isElementExist(element string, list []string) bool {
+	isExist := false
+	for j := 0; j < len(list); j++ {
+		if list[j] == element {
+			isExist = true
+		}
+	}
+	return isExist
+}
+
+// Grade Staking Price Records
+func (d *Pegnetd) GradeDelegatedS(ctx context.Context, block *factom.EBlock) (graderDelegateStake.DelegatedGradedBlock, error) {
+	if block == nil {
+		// TODO: Handle the case where there is no opr block.
+		// 		Must delay conversions if this- happens
+		return nil, nil
+	}
+
+	if *block.ChainID != config.SPRChain {
+		return nil, fmt.Errorf("trying to grade a non-spr chain")
+	}
 
-	g, err := graderStake.NewGrader(ver, int32(block.Height))
-	if err != nil {
-		return nil, err
+	ver := uint8(8)
+	if block.Height >= config.PIP18DelegateStakingActivation {
+		ver = 8
 	}
-	for _, entry := range block.Entries {
-		extids := make([][]byte, len(entry.ExtIDs))
-		for i := range entry.ExtIDs {
-			extids[i] = entry.ExtIDs[i]
+
+	if ver == 8 {
```

Comment:
ver is set to 8
You do a test, and if true, you set ver to 8
Then you test if ver is 8

ver is always 8, and the test is always true.

Reply:
That's right, it is for future version.
If we add PIP19, PIP20 activation heights, those versions would be 9, 10 for example.
`GradeDelegatedS()` will be used for future grading function as well.